src/test/java/org/openrewrite/staticanalysis/NeedBracesTest.java

```diff
@@ -323,6 +333,177 @@ static void method() {
                       }
                       return;
                   }
+                  static void commentWhile(Object obj) {
+                      while (true) { // while comment
+                      }
+                      while (true) {
+                          return; // while comment with body
+                      }
+                      do {
+                          obj.notify(); // do while comment
+                      } while (true);
+                  }
+                  static void commentIterative(Object obj) {
+                      for (int i = 0; ; ) { // iterative comment
+                      }
+                      for (int i = 0; ; ) {
+                          obj.notify(); // iterative with body comment
+                      }
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Issue("https://github.com/openrewrite/rewrite-static-analysis/issues/315")
+    @Test
+    void commentsBeforeElseBlockWithNoBraces() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              class Test {
+                  static void method() {
+                      if (true) {
+                          return;
+                      }
+                      /*
+                       * comment should be in else block
+                       */
+                      else return;
+                      if(true) {
+                          return;
+                      } // comment on if
+                      else return;
+                  }
+              }
+              """,
+            """
+              class Test {
+                  static void method() {
+                      if (true) {
+                          return;
+                      } else {
+                          /*
+                           * comment should be in else block
+                           */
+                          return;
+                      }
```

Comment:
Similarly, I had expected the following to minimize the diff with what was there before. All we ought to be doing is adding braces, where possible.
```suggestion
                      }
                      /*
                       * comment should be in else block
                       */
                      else {
                          return;
                      }
```